README.md

```diff
@@ -15,395 +15,18 @@ Project design document: [Design for a CoinSwap Implementation for Massively Imp
-    /// Protocol Handshake.
-    TakerHello(TakerHello),
-    /// Request the Maker's Offer advertisement.
-    ReqGiveOffer(GiveOffer),
-    /// Request Contract Sigs **for** the Sender side of the hop. The Maker receiving this message is the Receiver of the hop.
-    ReqContractSigsForSender(ReqContractSigsForSender),
-    /// Respond with the [ProofOfFunding] message. This is sent when the funding transaction gets confirmed.
-    RespProofOfFunding(ProofOfFunding),
-    /// Request Contract Sigs **for** the Receiver and Sender side of the Hop.
-    RespContractSigsForRecvrAndSender(ContractSigsForRecvrAndSender),
-    /// Request Contract Sigs **for** the Receiver side of the hop. The Maker receiving this message is the Sender of the hop.
-    ReqContractSigsForRecvr(ReqContractSigsForRecvr),
-    /// Respond with the hash preimage. This settles the HTLC contract. The Receiver side will use this preimage unlock the HTLC.
-    RespHashPreimage(HashPreimage),
-    /// Respond by handing over the Private Keys of coinswap multisig. This denotes the completion of the whole swap.
-    RespPrivKeyHandover(PrivKeyHandover),
-}
-```
-
-A step-by-step communication sequence with the above messages is provided in `src/messages.rs` [docs](https://github.com/utxo-teleport/teleport-transactions/blob/30be708642cfdaa206d52e147ecb580af7db0bda/src/messages.rs#L20-L59).
-
-The `Taker` carries out all the heavy lifting of the protocol. `Maker`s work like simple state-machine responding to `TakerToMakerMessage`s.
-
-`src/taker.rs` : describes the Taker protocol, which is the workflow defined in the [section above](#protocol-between-takers-and-makers). This is the core of the protocol implementation and the most security-critical section of the library.
-
-`src/maker.rs` : describes the Maker state-machine. This is a simple server responding to various `TakerToMakerMessage`s depending on a `ConnectionState`. Each `ConnectionState` will have specific messages as "allowed". The Maker will terminate the protocol if a received message doesn't match the allowed messages of a specific state.
-
-### Further reading
-
-* [Waxwing's blog post from 2017 about CoinSwap](https://web.archive.org/web/20200524041008/https://joinmarket.me/blog/blog/coinswaps/)
-
-* [gmaxwell's original coinswap writeup from 2013](https://bitcointalk.org/index.php?topic=321228.0). It explains how CoinSwap actually works. If you already understand how Lightning payment channels work then CoinSwap is similar.
-
-* [Design for improving JoinMarket's resistance to sybil attacks using fidelity bonds](https://gist.github.com/chris-belcher/18ea0e6acdb885a2bfbdee43dcd6b5af/). Document explaining the concept of fidelity bonds and how they provide resistance against sybil attacks.
-
+## Docs
+Check [how_to.md](docs/how_to.md) for steps on how to create a coinswap with this implementation.
 
+Check [Developer Resources](docs/developer_resources.md) on information on the protocol and further reading.
 
```

Comment:
Can we add build and test instructions here too? Just two lines of build the binary with `cargo build`, and run all the tests with `cargo test`. Its trivial, but just not written anywhere.

Reply:
Yep, added an "Install/Build from Source" section in the README and removed it from `app_instructions`